sphinx/builders/linkcheck.py

```diff
@@ -424,6 +425,7 @@ def check(docname: str) -> tuple[str, str, int]:
             check_request = self.wqueue.get()
             next_check, hyperlink = check_request
             if hyperlink is None:
+                self._session.close()
```

Comment:
Why do we close here?

Reply:
I wanted to ensure that the `requests` session's resources are cleaned up when the linkchecker completes -- some warnings appeared during testing to indicate that they weren't -- and this seemed to be a good place for it (the `None` hyperlink is used as a marker (sentinel?) to indicate that the worker should shutdown).